src/lib/bindings/http/http_emitter.ts

```diff
@@ -78,17 +85,20 @@ class HTTPEmitter {
  * @param {string} [version] spec version number - default 1.0
  * @returns {Object} the headers that will be sent for the event
  */
-function headers(event, version = SPEC_V1) {
+function headers(event: CloudEvent, version = SPEC_V1) {
   const headers = {};
   let headerMap;
   if (version === SPEC_V1) {
     headerMap = EmitterV1;
   } else if (version === SPEC_V03) {
     headerMap = EmitterV03;
   }
-  headerMap.forEach((parser, getterName) => {
+  headerMap.forEach((parser: any, getterName: string) => {
```

Comment:
What is the type of `parser`? We probably want to make the `Base64Parser` and `JSONParser` extend the same parser.

We probably don't need classes though, since these parsers should really be methods and simplified.

Reply:
I would prefer to open a new issue/pr for this. There are a bunch of changes that still could be done to the codebase to improve it overall - nits like `no-any` as you mention above as well as some of these bigger style/architecture concerns.